cli/CampaignConfigValidation/Rule/UniqueBucketRule.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Cli\CampaignConfigValidation\Rule;
+
+use WMDE\Fundraising\Frontend\BucketTesting\Campaign;
+use WMDE\Fundraising\Frontend\Cli\CampaignConfigValidation\ValidationErrorLogger;
+
+/**
+ * @license GNU GPL v2+
+ */
+class UniqueBucketRule {
+
+	public function validate( Campaign $campaign, ValidationErrorLogger $errorLogger ): bool {
+		$buckets = [];
+		$valid = true;
+		foreach ( $campaign->getBuckets() as $bucket ) {
+			if ( in_array( $bucket->getName(), $buckets ) ) {
```

Comment:
As a small performance optimization, you could use `$buckets` as a lookup table:

`isset( $buckets[$bucket->getName] )`